metadata.themoviedb.org.python/python/lib/tmdbscraper/tmdbapi.py

```diff
@@ -0,0 +1,132 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# pylint: disable=missing-docstring
+
+"""Functions to interact with TMDb API."""
+
+from . import api_utils
+import xbmc
+try:
+    from typing import Optional, Text, Dict, List, Any  # pylint: disable=unused-import
+    InfoType = Dict[Text, Any]  # pylint: disable=invalid-name
+except ImportError:
+    pass
+
+
+HEADERS = (
+    ('User-Agent', 'Kodi Movie scraper by Team Kodi'),
+    ('Accept', 'application/json'),
+)
+api_utils.set_headers(dict(HEADERS))
+
+TMDB_PARAMS = {'api_key': 'f090bb54758cabf231fb605d3e3e0468'}
+BASE_URL = 'https://api.themoviedb.org/3/{}'
+SEARCH_URL = BASE_URL.format('search/movie')
+FIND_URL = BASE_URL.format('find/{}')
+MOVIE_URL = BASE_URL.format('movie/{}')
+COLLECTION_URL = BASE_URL.format('collection/{}')
+CONFIG_URL = BASE_URL.format('configuration')
+
+
+def search_movie(query, year=None, language=None):
+    # type: (Text) -> List[InfoType]
+    """
+    Search for a movie
+
+    :param title: movie title to search
+    :param year: the year to search (optional)
+    :param language: the language filter for TMDb (optional)
+    :return: a list with found movies
+    """
+    xbmc.log('using title of %s to find movie' % query, xbmc.LOGDEBUG)
+    theurl = SEARCH_URL
+    params = _set_params(None, language)
+    params['query'] = query
+    if year is not None:
+        params['year'] = str(year)
+    return api_utils.load_info(theurl, params=params)
+
+
+def find_movie_by_external_id(external_id, language=None):
+    # type: (Text) -> List[InfoType]
+    """
+    Find movie based on external ID
+
+    :param mid: external ID
+    :param language: the language filter for TMDb (optional)
+    :return: the movie or error
+    """
+    xbmc.log('using external id of %s to find movie' % external_id, xbmc.LOGDEBUG)
+    theurl = FIND_URL.format(external_id)
+    params = _set_params(None, language)
+    params['external_source'] = 'imdb_id'
+    return api_utils.load_info(theurl, params=params)
+
+
+
+def get_movie(mid, language=None, append_to_response=None):
+    # type: (Text) -> List[InfoType]
+    """
+    Get movie details
+
+    :param mid: TMDb movie ID
+    :param language: the language filter for TMDb (optional)
+    :append_to_response: the additional data to get from TMDb (optional)
+    :return: the movie or error
+    """
+    xbmc.log('using movie id of %s to get movie details' % mid, xbmc.LOGDEBUG)
```

Comment:
Can you comment out all these log messages? Last I checked, we did not allow to log by default without a setting.

Reply:
Much like parley, that seems to be more of a guideline than a rule, but yes, I can do that.  '-)   This issue exists with both the not yet approved Leia version and the already approved Matrix version. Would you like me to fix for Matrix too and submit an update? BTW, It looks like I deleted a bunch of my forks, so I might have to resubmit the Leia PR anyway if we change anything.

BTW, In the TV scraper I do log a few things like this to debug (with no setting to turn it off), but I have a setting that only enables the API response dump with a setting because it dumps a metric crap ton of data to the log.

Reply:
I guess we need to talk about the guideline then @enen92 
Anyway, I won't block it for this, I would just prefer it that way. And it's fine to update the other code with the next changes you push (if you want to change the logging)

Reply:
The only rule we we've been enforcing is the usage of debug log level by default (and pretty much all the time unless really justified). The usage of a setting is sometimes advised but not mandatory. Should that be changed? If so we need to discuss this internally

Reply:
I don't know exactly what logging you are referring to, but please be aware that moderators need logging to assist users with scraping problems. @Razzeee I always wanted you to add more logging to your scraper, so please don't reduce logging to a point where the debuglog becomes useless.

Reply:
I also don't remember any hard rules regarding verbose debug (not info) logging.

Reply:
Yeah, the ones I remember are at least 5 years old and meant that the trakt addon needed to have a log setting and only use that, when debug logging in kodi is active too.
I'm pretty sure that @MartijnKaijser was involved, not so sure if @ronie was

Reply:
i do remember the same as @enen92, debug logging is mandatory for addons, but having a setting to enable logging isn't.